pkg/reloader/reloader.go

```diff
@@ -195,18 +202,24 @@ WaitForEvent:
 				// config files here and continue with our business.
 				break
 			}
-
 		case err := <-configWatcher.Errors:
 			log.Printf("Error: %s\n", err)
 			continue
 		}
 
+		// Reread process data, in case it changed - just to ensure to send the signal to
+		// the right process.
+		err = r.waitForProcess()
+		if err != nil {
+			return err
+		}
+
 		// Configuration was updated, try to do reload for a few times
 		// otherwise give up and wait for next event.
 	TryReload:
 		for {
-			log.Println("Sending signal to server to reload configuration")
-			err := r.proc.Signal(syscall.SIGHUP)
+			log.Printf("Sending signal '%s' to server to reload configuration\n", r.Signal.String())
```

Comment:
nb: `log` package prints include the trailing newline automatically (but at least won't duplicate one already present).

Reply:
Deleted all newlines from log messages.